README.md

```diff
@@ -1,12 +1,114 @@
+
+```
+{
+	number: 1,
+	solutions: [
+		“public int[] twoSum(int[] nums, int target) {.....”,
+		“public int[] twoSum(int[] nums, int target) {.....”,
+		“public int[] twoSum(int[] nums, int target) {.....”
+	],
+	problem_statement: “Given an array of integers, return indices of the two numbers…”
+}
+```
+
+These objects are stored in `problem.json`.
+
+
+### Labeling
+
+Relevant files: `labeler.py`, `problem.json`
+
+Labeling of problems statements is performed using `labellers.py` on `problems.json`. 
+For each problem, every solution of the problem is checked to see if it has keywords that indicate if it has a feature.
+These labels are stored as lists of integer values for each solution, where `0` indicates the feature is present, and `1` indicates that the feature is not present.,
+
+These functions are called in `classifer.py`. 
+
+
+### Classification
+
+Relevant files: `embedder.py`, `classifier,py`, `modeldata.py`
+
+#### Embedding 
+
+For each problem statement, a word embedding (mathematical representation of words as a vector of real numbers) is created using [Google's BERT models](https://towardsdatascience.com/nlp-extract-contextualized-word-embeddings-from-bert-keras-tf-67ef29f60a7b).
+The word embeddings are intended to capture the semantic meaning of each problem statement so problem statements that are semantically similar to each other will have vectors that are close to each other numerically.
+
+The functions to perform this embedding are defined in `embedder.py` and used in `classifier.py.`
+
+#### SVM (Support Vector Machine)
+
+This is a type of [supervised learning model](https://en.wikipedia.org/wiki/Supervised_learning) that can be used for classification.
+Support vector machines [create a hyperplane in multidimensional space](https://www.youtube.com/watch?v=3liCbRZPrZA&feature=emb_logo) to separate a dataset into classes.
+
+![Example of SVM](docs/svm-example.png)
+
+An ideal choice of hyperplane will have the largest possible distance (margin) between the two points of data that are closest to the hyperplane (support vectors).
+
+For this project, we use `scikit-learn`'s out-of-the-box [SVM classification model](https://scikit-learn.org/stable/modules/svm.html) to perform the classifications based on the problem statement labels and use [k-fold cross validation](https://en.wikipedia.org/wiki/Cross-validation_(statistics)#k-fold_cross-validation) to assess the accuracy of the model.
+
+
 ## Dependencies
 
-* Python 3
+* Python 3.8
 * [SciKit Learn](https://scikit-learn.org/stable/install.html)
 * [PyTorch](https://pytorch.org/get-started/locally/)
 * Bert Model: `pip3 install pytorch-pretrained-bert`
 * Numpy: `pip3 install numpy`
+
+## Run Instructions
+
+To run the classifier for all features, in the project's root directory, run `python3 src/classifier.py`. 
+
+### Running Model for One Feature
+
+To run the model for a specific feature, comment out the corresponding `gridsearch(...)` call in `classifier.py`.
```

Comment:
Instead of saying _"comment out the corresponding gridsearch(...) call"_, it may be better so say _"identify the corresponding gridsearch(...) call and comment out the others"_ since commenting out the corresponding gridsearch() would run all features except for the one desired.